eth/ethconfig/config.go

```diff
@@ -70,6 +70,7 @@ func NewDefaultConfig() Config {
 		RPCEVMTimeout:             5 * time.Second,
 		GPO:                       DefaultFullGPOConfig,
 		RPCTxFeeCap:               1, // 1 AVAX
+		RecentBlocksWindow:        1024,
```

Comment:
Can we plumb this configuration to control `tipBufferSize` here: https://github.com/ava-labs/coreth/blob/7a90507f3cbebfe85a65b4b26cf5383a87e951f3/core/state_manager.go#L49 as well?

Instead of being a const it can be taken in NewTrieWriter as a parameter or part of CacheConfig.

Reply:
Done, however, shouldn't this be its own option? 🤔 Doesn't it feel a bit decoupled from the recent blocks window allowed for GetProofs? Plus, what should it be if the node has `historical-blocks` set to true??

Reply:
Let's make it a separate option as you suggest for maximum flexibility.
In the future these may be tangled as with the pathDB the # of states we keep in memory will determine what amount of proofs can be served.

Reply:
I would suggest either:

1. Have the tip buffer internally being set to
  - a fixed number (`1024`?) if `historical-proofs` is `true`
  - the value of `recent-blocks-window` if `historical-proofs` is `false`
2. Have a new option `trie-tip-buffer` defaulting to
  - a fixed number (`1024`?) if `historical-proofs` is `true`
  - the value of `recent-blocks-window` if `historical-proofs` is `false`

I would use `1.` for now, since we can do `2.` later with retro compatibility if a user needs such flexibility. My take is to give the least options possible to the user (less cognitive load, less prone to errors). If a user requests it, we can provide the option.

Reply:
Since the tip buffer impacts memory usage, I would be cautious about a change from its current default of 32 --> 1024. This is why I prefer to either leave the tip buffer alone for now or to make it configurable separately without modifying its default value. In any case, if the API cannot access the state it will result in an error and the new setting of `historical-proofs` doesn't change that.

I think nodes that would serve historical proofs would also be archive nodes so the tip buffer is not applicable for them.

Let's also get thoughts from @ceyonur here.